### PR TITLE
Dp/invoices endpoint

### LINKED ISSUE #4: Invoices record endpoint

### PR DESCRIPTION
**Pull request by Dpalazzari**
Closes #4 

This branch:

- Tests and adds invoices API functionality for index and show endpoints.

- Tests and adds invoices API functionality for find and find_all endpoints.

- Test and adds invoices API functionality for random endpoint.

- Small tweek to spec folder file directory: made items folder structure consistent with all the other directories.

@sallymacnicholas @tmikeschu

### CODE REVIEW

**Comment by Dpalazzari** [comment]
Ready for review @tmikeschu @sallymacnicholas 🚀

File: app/controllers/api/v1/invoices/random_controller.rb [hunk 1]
```diff
@@ -0,0 +1,7 @@
+class Api::V1::Invoices::RandomController < ApplicationController
+
+	def show
+		render json: Invoice.random
+	end
```
Comment (Dpalazzari) [inline app/controllers/api/v1/invoices/random_controller.rb:5]: This is weird that Github shows the formatting this way. it is NOT like this in my local machine's code.

@sallymacnicholas @tmikeschu

Comment (Dpalazzari) [reply app/controllers/api/v1/invoices/random_controller.rb:5]: screen shot 2017-01-24 at 1 26 28 pm

**Comment by tmikeschu** [comment]
@Dpalazzari : looks great 👍 . One more endpoints for basic records!

**Pull request merged**